Sources/Fuzzilli/FuzzIL/TypeInfo.swift

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+public enum TypeQuality {
+    case infered
+    case runtime
+
+    public var rawValue: Int {
+        switch self {
+            case .infered: return 0
+            case .runtime: return 1
+        }
+    }
+
+    public init(rawValue: Int) {
+        switch rawValue {
+            case 0: self = .infered
+            case 1: self = .runtime
+            default: self = .infered
+        }
+    }
+}
+
+// Store known types for program variables at specific instructions
+public struct TypeInfo: Equatable {
+    public let index: Int
```

Comment:
Since we'll probably have quite a lot of these structs in memory, let's make it
```
public let index: UInt16 // Guaranteed to fit into a UInt16, see Instruction.swift
public let quality: TypeQuality // rawValue is UInt16 or UInt8
public let type: Type
```